File: programmers/network/network.java
class Solution {
    static int ans = 0;
    static boolean[] visited;

    public void sol1(int index, int n, int[][] computers){

        visited[index] = true;

        for(int i = 0; i < n ; i++){
            if(visited[i] != true && computers[index][i] == 1) {
                sol1(i, n, computers);
            }
        }
    }

    public int solution(int n, int[][] computers) {
        visited = new boolean[n];

        for(int i = 0 ; i < n ; i++){
            if(visited[i] != true){
                ans++;
                sol1(i, n, computers);
            }
        }

        return ans;
    }
}

File: programmers/network/network.js
function solution(n, computers) {
  const visited = Array(n);
  [...Array(n).keys()].forEach(i => visited[i] = false);
  
  let count = 0;
  for(let i = 0; i < n ; i++){
      if(visited[i]) continue;
      
      count++;
      const stock = [i];
      
      while(stock.length > 0){
          const node = stock.pop();
          if(visited[node]) continue;
          
          visited[node] = true;
          computers[node][node] = 0;
          computers[node].forEach((next, idx) => {
              if(next === 1){
                  stock.push(idx);
              }
          });
      }
  }
  
  return count;
}
